### PR TITLE
feat(talk-details): design and screen structure

### PR DESCRIPTION
**Pull request by frankcalise**
# Description

[Trello Card](36-screen-talk-details)

Implements the `TalkDetailsScreen`
- Animated the title to show in the navigation bar when the text labels in the scroll view roll under the navigation bar
- Follows design from Figma
> **Note**
> - This is just the Screen component with hardcoded data, it still needs to be wired up to the shape of the model from the Schedule screen and pass the data
> - `isWorkshop` boolean placeholder to style between workshop/talk style speaker images and shapes to be wired up to model shape when we know more

# Graphics

| iOS | Android |
| ------ | ------- |
| Simulator Screen Recording - iPhone 13 - 2022-12-02 at 09 17 37 | |
|  |  |
|  |  |
|  |  |

# Checklist:

- [x] I have done a thorough self-review of my code
- [ ] I have tested with a screen reader and font-scaling turned on and added necessary accessibility features 
- [x] I have run tests and linter

### CODE REVIEW

File: app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+      },
+    })
+
+    const { bottom: paddingBottom } = useSafeAreaInsets()
+
+    // TODO: wire this up to the event type
+    const isWorkshop = true
+
+    return (
+      <SafeAreaView edges={["top", "bottom"]} style={$root}>
```
Comment (jonmajorc) [inline app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx:39]: Curious on why are you not using the the `Screen` component?

**Review by jonmajorc — COMMENTED** [review]
Looks good on my end! Not sure about these zIndexes how they will scale 🤔

File: app/screens/WelcomeScreen.tsx [hunk 2]
```diff
@@ -86,7 +86,6 @@ const $bottomContentContainer: ViewStyle = Platform.select({
   },
   android: {
     flex: 1,
-    width: screenWidth * 0.5,
```
[the remark this answers is not in the document]
Comment (frankcalise) [reply app/screens/WelcomeScreen.tsx:89]: When I ran Android the button wasn't centered anymore, but centered within the left half of the screen.

File: app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx (hunk 1 above)
Comment (frankcalise) [reply app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx:39]: Hmm I will check that, I had removed it at one point playing with different animation code so I might just have forgot to revert it back. Let me look

**Comment by frankcalise** [comment]
> Looks good on my end! Not sure about these zIndexes how they will scale 🤔

@jonmajorc yeah I agree, do you have a better way of approaching that styling property when it comes into play? Like always bump by 5 or 10 so there is room for insertion or something like that?

File: app/screens/TalkDetailsScreen/TalksDetailsHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  justifyContent: "space-between",
+}
+
+const $centerTitle: TextStyle = {
+  position: "absolute",
+  width: "100%",
+  textAlign: "center",
+  paddingHorizontal: spacing.huge,
+  zIndex: 1,
+  fontSize: 16,
```
Comment (mazenchami) [inline app/screens/TalkDetailsScreen/TalksDetailsHeader.tsx:59]: [TOL] instead of a static size, can we use a `preset`?

Comment (frankcalise) [reply app/screens/TalkDetailsScreen/TalksDetailsHeader.tsx:59]: Umm I would but it isn't our Ignite `Text` component, it's from `<Animated.Text />`. Is this where you would use `Animated.createAnimatedComponent`?

Comment (mazenchami) [reply app/screens/TalkDetailsScreen/TalksDetailsHeader.tsx]: Yeah that's what I would do! Looks good to me

**Review by mazenchami — APPROVED** [review]
LGTM! 🎉

**Pull request merged**